File: pdlearning/src/web-apps/src/libraries/domain-api/src/question-bank/dtos/save-question-bank-request.ts
import { QuestionAnswerValue, QuestionType } from '../../form/models/form-question.model';

import { QuestionBank } from '../models/question-bank';
import { QuestionOption } from '../../form/models/form-question-option.model';

export interface ISaveQuestionBankRequest {
  id?: string;
  title: string;
  questionGroupName: string;
  questionTitle: string;
  questionType: QuestionType;
  questionCorrectAnswer: QuestionAnswerValue;
  questionOptions: QuestionOption[];
  questionHint: string | null;
  answerExplanatoryNote: string | undefined;
  feedbackCorrectAnswer: string | undefined;
  feedbackWrongAnswer: string | undefined;
  questionLevel: number | undefined;
  randomizedOptions: boolean | undefined;
  score: number | undefined;
  isDeleted: boolean;
  isScoreEnabled?: boolean;
}

export class SaveQuestionBankRequest {
  public id?: string;
  public title: string;
  public questionGroupName: string;
  public questionTitle: string;
  public questionType: QuestionType;
  public questionCorrectAnswer: QuestionAnswerValue;
  public questionOptions: QuestionOption[];
  public questionHint: string | null;
  public answerExplanatoryNote: string | undefined;
  public feedbackCorrectAnswer: string | undefined;
  public feedbackWrongAnswer: string | undefined;
  public questionLevel: number | undefined;
  public randomizedOptions: boolean | undefined;
  public score: number | undefined;
  public isDeleted: boolean;
  public isScoreEnabled?: boolean;

  constructor(data: QuestionBank) {
    if (data === undefined) {
      return;
    }

    this.id = data.id;
    this.title = data.title;
    this.questionGroupName = data.questionGroupName;
    this.questionType = data.questionType;
    this.questionTitle = data.questionTitle;
    this.questionCorrectAnswer = data.questionCorrectAnswer;
    this.questionOptions = data.questionOptions;
    this.questionHint = data.questionHint;
    this.answerExplanatoryNote = data.answerExplanatoryNote;
    this.feedbackCorrectAnswer = data.feedbackCorrectAnswer;
    this.feedbackWrongAnswer = data.feedbackWrongAnswer;
    this.questionLevel = data.questionLevel;
    this.randomizedOptions = data.randomizedOptions;
    this.score = data.score;
    this.isDeleted = data.isDeleted;
    this.isScoreEnabled = data.isScoreEnabled;
  }
}

File: pdlearning/src/microservices/form-microservice/Api/Application/RequestDtos/QuestionBank/SaveQuestionBankRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microservice.Form.Application.Commands;
using Microservice.Form.Application.Models;
using Microservice.Form.Domain.ValueObjects.Questions;

namespace Microservice.Form.Application.RequestDtos
{
    public class SaveQuestionBankRequest
    {
        public Guid? Id { get; set; }

        public string Title { get; set; }

        public string QuestionGroupName { get; set; }

        public string QuestionTitle { get; set; }

        public QuestionType QuestionType { get; set; }

        public object QuestionCorrectAnswer { get; set; }

        public IEnumerable<QuestionOptionModel> QuestionOptions { get; set; } = new List<QuestionOptionModel>();

        public string QuestionHint { get; set; }

        public string QuestionAnswerExplanatoryNote { get; set; }

        public string QuestionFeedbackCorrectAnswer { get; set; }

        public string QuestionFeedbackWrongAnswer { get; set; }

        public int? QuestionLevel { get; set; }

        public bool? RandomizedOptions { get; set; }

        public double? Score { get; set; }

        public bool IsDeleted { get; set; }

        public bool IsScoreEnabled { get; set; }

        public SaveQuestionBankCommand ToCommand(Guid userId, bool isCreation = false)
        {
            return new SaveQuestionBankCommand
            {
                Id = Id.HasValue ? Id.Value : Guid.NewGuid(),
                IsDeleted = IsDeleted,
                IsScoreEnabled = IsScoreEnabled,
                QuestionAnswerExplanatoryNote = QuestionAnswerExplanatoryNote,
                QuestionCorrectAnswer = QuestionCorrectAnswer,
                QuestionFeedbackCorrectAnswer = QuestionFeedbackCorrectAnswer,
                QuestionFeedbackWrongAnswer = QuestionFeedbackWrongAnswer,
                QuestionGroupName = QuestionGroupName,
                QuestionHint = QuestionHint,
                QuestionLevel = QuestionLevel,
                QuestionOptions = QuestionOptions,
                QuestionTitle = QuestionTitle,
                QuestionType = QuestionType,
                RandomizedOptions = RandomizedOptions,
                Score = Score,
                Title = Title,
                IsCreation = isCreation,
                UserId = userId
            };
        }
    }
}
